apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment_school_info.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import {Table} from 'react-bootstrap';
+import ConfirmationDialog from './confirmation_dialog';
+import {enrollmentShape} from "../types";
+import {workshopEnrollmentStyles as styles} from "../workshop_enrollment_styles";
+
+export default class WorkshopEnrollmentSchoolInfo extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      pendingDelete: null
+    };
+
+    this.handleClickDelete = this.handleClickDelete.bind(this);
+    this.handleDeleteCanceled = this.handleDeleteCanceled.bind(this);
+    this.handleDeleteConfirmed = this.handleDeleteConfirmed.bind(this);
+  }
+
+  handleClickDelete(event) {
+    this.setState({
+      pendingDelete: {
+        id: event.currentTarget.dataset.id,
+        email: event.currentTarget.dataset.email,
+        first_name: event.currentTarget.dataset.first_name,
+        last_name: event.currentTarget.dataset.last_name
+      }
+    });
+  }
+
+  handleDeleteCanceled() {
+    this.setState({
+      pendingDelete: null
+    });
+  }
+
+  handleDeleteConfirmed() {
+    const pendingDeleteId = this.state.pendingDelete.id;
+    this.setState({
+      pendingDelete: null
+    });
+    this.props.onDelete(pendingDeleteId);
+  }
+
+  render() {
+    const enrollmentRows = this.props.enrollments.map((enrollment, i) => {
+      let deleteCell;
+      if (enrollment.attended) {
```

Comment:
Alternative to consider - give them the button but disable it. Maybe have a tooltip telling them to contact us if they need to delete this enrollment? I don't know how often this comes up

Reply:
Note this part is [unchanged](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment.jsx#L86-L88). The entire existing enrollment view was just moved into this new class so it could be reused.

Also the visual clue has actually been helpful (AFAIK). At least we haven't heard complaints about it. Only enrollments that can be deleted (i.e. have not attended) even have the minus button:

![image](https://user-images.githubusercontent.com/8405316/29984306-a5276a92-8f0e-11e7-9e5f-17891c42e170.png)